### PR TITLE
fix(k8s): compareDeployedResources & manifest hash annotations

### PR DESCRIPTION
**Pull request by stefreak**
**What this PR does / why we need it**:

* **fix(k8s): compareDeployedResources should not ignore manifest hash**

compareDeployedResources actually just skipped isSubset check in case
the manifest hash was present.

Now it actually returns "outdated" when manifest hash differs.

This is necessary if the manifest removes annotations that are
present in the deployed manifest.

E.g.:

1. upsert annotations: {"some-annotation": "present"}
1. upsert annotations: {}

before this change, compareDeployedResources just returned "ready"
even though the deployedManifest was actually outdated (the annotation
needed to be removed)

This commit fixes that

* **fix(k8s): add manifest-hash annotations also for upsert operations**

previously we added garden manifest-hash only for kubectl apply ops.

This commit makes sure to add the manifest-hash also for upserts.

This enables for more accurate (and quicker) comparison results in
compareDeployedResources for the affected resources.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: core/src/plugins/kubernetes/status/status.ts [hunk 1]
```diff
@@ -381,15 +381,28 @@ export async function compareDeployedResources(
     if (deployedResource.metadata && deployedResource.metadata.annotations) {
       const lastAppliedHashed = deployedResource.metadata.annotations[gardenAnnotationKey("manifest-hash")]
 
-      // The new manifest matches the last applied manifest
-      if (lastAppliedHashed && (await hashManifest(manifest)) === lastAppliedHashed) {
-        continue
+      if (lastAppliedHashed) {
+        const calculated = await hashManifest(manifest)
+        if (calculated !== lastAppliedHashed) {
```
Comment (stefreak) [inline core/src/plugins/kubernetes/status/status.ts:386]: Probably we should keep the continue if it's equal – that way we are saving some cpu cycles in that case.

Comment (stefreak) [reply core/src/plugins/kubernetes/status/status.ts:386]:
```suggestion
        if (calculated === lastAppliedHashed) {
        continue
        } else {
```

File: core/src/plugins/kubernetes/api.ts [hunk 2]
```diff
@@ -597,6 +597,17 @@ export class KubeApi {
     obj: O
     log: LogEntry
   }) {
+    // Hash the raw input and add as an annotation on each manifest (this is helpful beyond kubectl's own annotation,
```
Comment (stefreak) [inline core/src/plugins/kubernetes/api.ts:600]: this code block is copied from `apply()` and should be extracted to a helper method

**Comment by vvagaytsev** [comment]
There are some test failures. I'll convert this to a draft, and will try to take a look later this week.

**Pull request closed without merging**